docs/collection.js

```diff
@@ -0,0 +1,279 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Abstract base class containing methods shared by {@link Realm.List} and {@link Realm.Results}.
+ * @memberof Realm
+ * @since 0.11.0
+ */
+class Collection {
+    /**
+     * The number of objects in the results.
+     * @type {number}
+     * @readonly
+     */
+    get length() {}
+
+    /**
+     * Returns new results that represent this collection being filtered by the provided query.
+     * @param {string} query - Query used to filter results.
+     * @param {...any} [arg] - Each subsequent argument is used by the placeholders
+     *   (e.g. `$0`, `$1`, `$2`, …) in the query.
+     * @throws {Error} If the query or any other argument passed into this method is invalid.
+     * @returns {Realm.Results} filtered according to the provided query.
+     * @example
+     * let merlots = wines.filtered('varietal == "Merlot" && vintage <= $0', maxYear);
```

Comment:
That example was there before, but I'm happy to use an alternative if you like. Please suggest one. :smile: